lib/semantics/check-return.cc

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "check-return.h"
+#include "semantics.h"
+#include "tools.h"
+#include "../parser/message.h"
+#include "../parser/parse-tree.h"
+
+namespace Fortran::semantics {
+
+const Scope *FindContainingSubprogram(const Scope &start) {
+  const Scope *scope{&start};
+  while (scope != nullptr) {
```

Comment:
scope is only ever set to the address of a Scope, so it can never be null.

The condition should be `scope->kind() != Scope::Kind::Global`.

Reply:
OK.  I was following FindProgramUnitContaining, which has the same while loop.